src/kudu/consensus/consensus_peers.cc

```diff
@@ -292,9 +292,10 @@ void Peer::SendNextRequest(bool even_if_queue_empty, bool from_heartbeater) {
   string next_hop_uuid;
   int64_t commit_index_before = request_.has_committed_index() ?
       request_.committed_index() : kMinimumOpIdIndex;
-  Status s = queue_->RequestForPeer(peer_pb_.permanent_uuid(), read_ops, &request_,
-                                    &replicate_msg_refs_, &needs_tablet_copy,
-                                    &next_hop_uuid);
+  bool enforce_response_token;
+  Status s = queue_->RequestForPeer(
+      peer_pb_.permanent_uuid(), read_ops, &request_, &replicate_msg_refs_,
+      &needs_tablet_copy, &next_hop_uuid, &enforce_response_token);
```

Comment:
why is enforce_response_token a return output here. At the time of start of request, i.e. before RequestForPeer we should know whether we want to enforce or not?

Reply:
We actually don't, because the peer object is created once per known peer and it's state is not tied to any request.

This is why we call back into the queue to ask for a request to be built for the peer. Whether we enforce the token is also a request state, which is why it has to be returned here.